### PR TITLE
Support Opset20 for IsInf and IsNaN

### PR DESCRIPTION
**Pull request by hamptonm1**
I updated onnx.IsInf and onnx.IsNaN to be compatible with opset 20. The updates for both of these ops is supporting new input data types such as float16 and bfloat16. So I made the applicable changes in the code base as well as the documentation so that these ops are backwards compatible.

### CODE REVIEW

File: src/Accelerators/NNPA/Runtime/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,6 @@ set_target_properties(RuntimeNNPA
   PROPERTIES
   LANGUAGE C
   POSITION_INDEPENDENT_CODE TRUE
-  COMPILE_OPTIONS "-O3;-fopenmp"
+  COMPILE_OPTIONS -option O3 fopenmp
```
Comment (hamptonm1) [inline src/Accelerators/NNPA/Runtime/CMakeLists.txt:24]: @chentong319 I added the `-option` to bypass this error I was seeing when compiling `onnx-mlir`.

`clang: error: unsupported option '-fopenmp'`

I got the idea to use option from the following documentation: https://cmake.org/cmake/help/latest/command/target_compile_options.html

Comment (hamptonm1) [reply src/Accelerators/NNPA/Runtime/CMakeLists.txt:24]: Or I can remove `-fopenmp` .... it is blocking my building

Comment (chentong319) [reply src/Accelerators/NNPA/Runtime/CMakeLists.txt:24]: Yes, you can remove fopenmp flag now.
Which build failed with this flag? This flag will be needed in future.

**Comment by chentong319** [comment]
Is it true that the latest version of IsInf and IsNaN allow more data type than the previous ones? Is there any other difference? If the latest version only adds some element types, we can use the newest op for the older versions. The only loss is the precise element type check in verifier, if there is a such check.

**Comment by hamptonm1** [comment]
> Is it true that the latest version of IsInf and IsNaN allow more data type than the previous ones? Is there any other difference? If the latest version only adds some element types, we can use the newest op for the older versions. The only loss is the precise element type check in verifier, if there is a such check. Multi-versioning of ops should be avoid as much as possible.

@chentong319 Yes this is true, I can do that then. Let me update the code. Thanks

File: src/Dialect/ONNX/ONNXOps/Math/ElementwiseUnary.cpp [hunk 2]
```diff
@@ -326,8 +326,8 @@ LogicalResult ONNXIsInfOp::inferShapes(
 
 LogicalResult ONNXIsNaNOp::inferShapes(
     std::function<void(Region &)> doShapeInference) {
-  IntegerType i1Type = IntegerType::get(getContext(), 1, IntegerType::Signless);
-  return inferShapeForUnaryOps(getOperation(), i1Type);
+  return inferShapeForUnaryOps(this->getOperation(),
+      this->getResult().getType().cast<ShapedType>().getElementType());
```
Comment (chentong319) [inline src/Dialect/ONNX/ONNXOps/Math/ElementwiseUnary.cpp:330]: Can it be guaranteed the output type is ShapedType?

Comment (hamptonm1) [reply src/Dialect/ONNX/ONNXOps/Math/ElementwiseUnary.cpp:330]: I can change It back to be safe... no guarantee

**Review by chentong319 — APPROVED** [review]
LGTM!

**Pull request merged**